### ISSUE TITLE
Prometheus empty string label value is not replaced in series legend templates

### ISSUE THREAD

**Issue by sandersaares** [grafana/grafana#19410]
**What happened**:
Say I have a Prometheus database that counts how many people enter my shop and what sort of car they arrived with.

```
persons_entered_shop_total{car = "Audi"} 5
persons_entered_shop_total{car = "Volkswagen"} 4
persons_entered_shop_total{car = ""} 8
```

Empty string indicates they did not come by car. Perfectly valid data set as far as Prometheus is concerned.

Now I make a graph of this and set `{{car}}` as my series legend name.

The result is the following series labels:

* Audi
* Volkswagen
* **car**

**What you expected to happen**:

I expected the 3rd series label to match the label value in Prometheus (empty string).

Ideally, I could also set an override for the empty string in Grafana graph settings but this is unrelated feature request - I certainly don't expect the replacement to just not happen.

**How to reproduce it (as minimally and precisely as possible)**:

Create Prometheus data set above and configure a graph with the series label string as above.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.3.6
- Data source type & version: Prometheus 2.12.0
- OS Grafana is installed on: Ubuntu (container)
- User OS & Browser: Firefox 69 on Windows 10
- Grafana plugins:
- Others:

**Comment by sandersaares** [comment]
It appears that Prometheus does not even return empty labels in its results. Regardless, this report is about the formatting issue, although I can see how Prometheus not even returning the label can make proper handling more complicated in Grafana.

**Comment by marefr** [comment]
> I can see how Prometheus not even returning the label can make proper handling more complicated in Grafana. 

Probably yes. But if prometheus not returning a value for the label Grafana should know that something that was expected is not there. Let's leave this open and see if others find this feature interesting.

**Comment by marefr** [comment]
I guess it's a simple workaround and that is to write the label value N/A or unknown so Prometheus can scrape and store that.

**Comment by praseodym** [comment]
Relabelling at the source could be pretty complicated when running third-party products or exporters. I was expecting there to be some syntax like `{{label_name|default value}}` but apparently there is none, so +1 for this feature request.

**Comment by brian-brazil** [comment]
Just came across this myself, this is a bug on the part of Grafana. A missing label is the same as an empty label in Prometheus, so if I do `{{unknownLabel}}` I expect the empty string as a result.

**Comment by oxddr** [comment]
I am also affected by this bug/feature. I don't have a control over metric's labels, so it's not possible to sensible format a legend for metrics that have some labels empty and some set.

**Comment by felmouatassim** [comment]
I have just faced the issue also. Since a missing label is the same as an empty label in Prometheus, then an empty result in Grafana would be the correct result.

**Comment by ivanahuckova** [comment]
I looked into this issue and it looks like Prometheus datasource in Grafana is doing a correct thing and sets up the name to be empty string. However, we overwrite it in [**calculateFieldDisplayName**](https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/field/fieldState.ts#L52) that is used in **getFieldDisplayName** that is used in [**getSeriesList**](https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/data_processor.ts#L23) that is used to render Graph and legend.

So what we currently do  - if the legend would be empty string, we show all of its labels + values. Such as here (where some of the series don't have boolean label):
![image](https://user-images.githubusercontent.com/30407135/106644603-920b3680-658b-11eb-8300-623e5d0e74ee.png)

But I am curious, isn't that a better option than showing couple of series with empty legend? Such as this (don't mind the aligning):
![image](https://user-images.githubusercontent.com/30407135/106644802-dc8cb300-658b-11eb-9f96-0014336af54b.png)

**Comment by aocenas** [comment]
Yeah having it empty in the legend looks pretty weird. We could have some specific "empty" placeholder there like "n/a" or "[empty]" or something like that. If this was done in the prometheus data source it won't affect other ones as there it could make sense to have that default.
